src/outputs/parthenon_hdf5.cpp

```diff
@@ -589,6 +589,9 @@ void PHDF5Output::WriteOutputFileImpl(Mesh *pm, ParameterInput *pin, SimTime *tm
                 global_count, pl_xfer);
 
     const H5G g_var = MakeGroup(g_swm, "SwarmVars");
+    if (swinfo.global_count == 0) {
+      continue;
+    }
```

Comment:
Just double checking that `global_count` is set correctly here, isn't it?
I'm asking as the `SetCounts` function (which also sets a `global_count [array]) follows afterwards.

Reply:
Yeah, the naming could use work, my apologies for that. The value in the `SwarmInfo` struct is total particle number and not hdf5 related:
https://github.com/parthenon-hpc-lab/parthenon/blob/develop/src/outputs/output_utils.cpp#L94